packages/backend/src/peripherals/coingecko/CoingeckoQueryService.ts

```diff
@@ -0,0 +1,33 @@
+import { CoingeckoClient, UnixTime } from "@l2beat/common";
+
+export enum Granularity {
+    Hourly,
+    Daily
+}
+
+export class CoingeckoQueryService {
+    constructor(
+       private coingeckoClient: CoingeckoClient
+    ){}
+
+}
+
+export function generateTimestampsList(
+    from: UnixTime,
+    to: UnixTime,
+    granularity: Granularity
+): Date[] {
```

Comment:
I think it makes sense for the function to return `UnixTime[]`. Let the user of this function worry about the conversion to milliseconds. If `UnixTime` is our preferred time structure in the system then I think we should hide the fact that coingecko returns milliseconds as an implementation detail of `CoingeckoQueryService`